### PR TITLE
Fix - `datetime.timedelta` to string conversion for JSON

### LINKED ISSUE #42: `TypeError` raised when JSON serialising `datetime.timedelta` object
I've been going through the documentation and encountered this when trying out a training run.

```python
[2024-09-02 14:09:06,632][lightning_fabric.utilities.seed][INFO] - [rank: 0] Seed set to 34257828
Error executing job with overrides: []
Traceback (most recent call last):
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/train/train.py", line 355, in main
    AnemoiTrainer(config).train()
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/train/train.py", line 342, in train
    self.model,
    ^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/micromamba/envs/aienv/lib/python3.11/functools.py", line 1001, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/train/train.py", line 139, in model
    "metadata": self.metadata,
                ^^^^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/micromamba/envs/aienv/lib/python3.11/functools.py", line 1001, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/train/train.py", line 212, in metadata
    return map_config_to_primitives(
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/utils/jsonify.py", line 50, in map_config_to_primitives
    config = {k: map_config_to_primitives(v) for k, v in config.items()}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/utils/jsonify.py", line 50, in <dictcomp>
    config = {k: map_config_to_primitives(v) for k, v in config.items()}
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/utils/jsonify.py", line 50, in map_config_to_primitives
    config = {k: map_config_to_primitives(v) for k, v in config.items()}
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/utils/jsonify.py", line 50, in <dictcomp>
    config = {k: map_config_to_primitives(v) for k, v in config.items()}
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi-training/src/anemoi/training/utils/jsonify.py", line 59, in map_config_to_primitives
    raise TypeError(msg)
TypeError: Cannot serialize object of type <class 'datetime.timedelta'>
```

My config file is 

```yaml
defaults:
  - data: zarr
  - dataloader: native_grid
  - diagnostics: eval_rollout
  - hardware: example
  - graph: multi_scale
  - model: gnn
  - training: default
  - _self_

data:
  resolution: n320

hardware:
  paths:
    output: /etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi/training/output
    data: /home/mlx/ai-ml/datasets/
    graph: /etc/ecmwf/nfs/dh1_home_a/ecm3786/Documents/Devel/anemoi/training/graphs
  files:
    dataset: aifs-od-an-oper-0001-mars-n320-2019-2021-6h-v4.zarr
    graph: "first_graph_m320.pt"
```

### PR DESCRIPTION
**Pull request by gareth-j**
This adds the conversion of a `datetime.timedelta` used for the frequency to a string for JSON serialisation.

Closes #42

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-training?pullRequest=43) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-training?pullRequest=43) before we can accept your contribution.<br/><hr/>**Gareth Jones** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/anemoi-training?pullRequest=43) it.</sub>

**Comment by gmertes** [comment]
Which part of the metadata now has a timedelta? This PR looks good to me but just trying to understand where the bug suddenly comes from, some new functionality in one of the downstream anemois maybe?

**Comment by gareth-j** [comment]
This [this PR in anemoi datasets](https://github.com/ecmwf/anemoi-datasets/pull/23/files) in introduced changes to the handling of frequency and the use of `frequency_to_timedelta` from `anemoi-utils`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ Keep it human-readable, your future self will thank you!
 - Enforce same binning for histograms comparing true data to predicted data
 - Fix: Inference checkpoints are now saved according the frequency settings defined in the config
 
+### Fixed
+
+- Fix `TypeError` raised when trying to JSON serialise `datetime.timedelta` object
```
Comment (mchantry) [inline CHANGELOG.md:23]: @gareth-j could you add  #43 to this line, to link to the PR please. This will be a new part of the style.

**Comment by mchantry** [comment]
Should the CI pipeline have flagged the anemoi-datasets change as introducing this?

File: CHANGELOG.md (hunk 1 above)
Comment (gareth-j) [reply CHANGELOG.md]: Done.

**Review by gmertes — COMMENTED** [review]
> Should the CI pipeline have flagged the anemoi-datasets change as introducing this?

The bug will only manifest during training, which we are not testing yet. If we had an integration test in the pipeline that loaded a real dataset and trained a real model, yes the CI could have flagged it. 

LGTM, but pinging @JesperDramsch because I think this touches their work and we briefly discussed revisiting this code in the future (maybe move it to utils?).

**Review by b8raoult — APPROVED** [review]

**Review by JesperDramsch — APPROVED** [review]
LGTM

**Pull request merged**